sql/core/src/test/scala/org/apache/spark/sql/execution/command/v2/AlterTableAddPartitionSuite.scala

```diff
@@ -68,4 +68,35 @@ class AlterTableAddPartitionSuite
       checkAnswer(sql(s"SELECT * FROM $t"), Seq(Row(0), Row(1)))
     }
   }
+
+  test("SPARK-34099, SPARK-34161: keep dependents cashed after table altering") {
```

Comment:
BTW can we unify this test between v1 and v2?

Reply:
At the moment, we cannot because v1 uncaches all dependants. At least now.

Reply:
And we cannot unify v1 and v2 `ALTER TABLE .. ADD PARTITION` because:
- v2 doesn't respect `LOCATION` but support fully partitioned table
- v1 is opposite. It supports `LOCATION` but not fully partitioned table.